fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryTest.java

```diff
@@ -0,0 +1,175 @@
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.directory;
+
+import com.apple.foundationdb.Transaction;
+import com.apple.foundationdb.record.RecordMetaData;
+import com.apple.foundationdb.record.RecordMetaDataBuilder;
+import com.apple.foundationdb.record.TestRecords1Proto;
+import com.apple.foundationdb.record.provider.foundationdb.FDBDatabase;
+import com.apple.foundationdb.record.provider.foundationdb.FDBDatabaseFactory;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordContext;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStore;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreTestBase;
+import com.apple.foundationdb.record.provider.foundationdb.TestKeySpace;
+import com.apple.foundationdb.subspace.Subspace;
+import com.apple.test.Tags;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.nio.file.NoSuchFileException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * Test for FDBDirectory validating it can function as a backing store
+ * for Lucene.
+ */
+@Tag(Tags.RequiresFDB)
+public class FDBDirectoryTest extends FDBRecordStoreTestBase {
+    private FDBRecordStore recordStore;
+    private FDBDatabase fdb;
+    private Subspace subspace;
+    private FDBDirectory directory;
+
+    @BeforeEach
+    public void setUp() {
+        if (fdb == null) {
+            fdb = FDBDatabaseFactory.instance().getDatabase();
+        }
+        if (subspace == null) {
+            subspace = fdb.run(context -> TestKeySpace.getKeyspacePath("record-test", "unit", "indexTest", "version").toSubspace(context));
+        }
+
+        fdb.run(context -> {
+            FDBRecordStore.deleteStore(context, subspace);
+            return null;
+        });
+        RecordMetaDataBuilder metaDataBuilder = RecordMetaData.newBuilder().setRecords(TestRecords1Proto.getDescriptor());
+        FDBRecordContext context = fdb.openContext();
+        recordStore = FDBRecordStore.newBuilder()
+                .setMetaDataProvider(metaDataBuilder)
+                .setContext(context)
+                .setSubspace(subspace)
+                .setFormatVersion(FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION)
+                .createOrOpen();
+        Transaction transaction = recordStore.ensureContextActive();
+        directory = new FDBDirectory(subspace, transaction);
+    }
+
+    @Test
+    public void testDirectoryCreate() {
+        assertNotNull(directory);
```

Comment:
Nitpick, but can we add an error message to these asserts? Makes it easier to track down precisely what went wrong when the tests fail.

Reply:
Fixing...